src/plugins/intel_gpu/src/graph/program_node.cpp

```diff
@@ -378,14 +378,40 @@ bool program_node::has_padded_dependency() const {
     });
 }
 
-std::map<size_t, memory::ptr> program_node::get_const_memory_deps() const {
+bool program_node::is_fused_dep(const primitive_id& dep_id) const {
+    for (auto& fused : get_fused_primitives()) {
+        for (auto& fused_dep : fused.deps) {
+            auto compare_rt = fused_dep.first.compare(dep_id);
+            if (compare_rt == 0) {
+                return true;
+            } else if (compare_rt < 0) {
+                auto cropped_dep_id = dep_id.substr(0, fused_dep.first.size());    // in case of adding '_reorder_'
+                if (fused_dep.first.compare(cropped_dep_id) == 0)
+                    return true;
+            }
+        }
+    }
+
+    return false;
+}
+
+std::map<size_t, memory::ptr> program_node::get_const_memory_deps(bool exclude_fused_dep) const {
     std::map<size_t, memory::ptr> mem_deps;
     for (auto& i : get_shape_infer_dependencies()) {
         // Some primitives may have flexible count of deps (e.g. reshape), thus allow skipping some deps
         if (i >= get_dependencies().size())
             continue;
```

Comment:
Can't the same to be achieved by 
```
if (i >= get_primitive()->input_size())
    continue;
```

?
get_dependencies() returns all deps including fused ones, while get_primitive()->input_size() inputs count of original primitive.

Reply:
I think this is better idea than using fused_primitive_desc::dep_start_idx

Reply:
I've changed to use fused_primitive_desc::dep_start_idx instead of get_primitive()->input_size() because get_primitive() doesn't represent original prim.